lib/query.js

```diff
@@ -232,7 +232,7 @@ export const getCoefficientsAndConstants = (node, coefficientMap = {}, constants
         } else {
             coefficientMap[key].push(coefficient)
         }
-    } else if(isPolynomial(node) || isApply(node)) {
+    } else if(isPolynomial(node) || isAdd(node) || (isMul(node) && node.args.every(isPolynomialTerm))) {
         // 2x^2 + 3x + 1, 3x^2 * 2x^2
```

Comment:
I don't think so; it leaves out non-polynomial expressions such as `x + 4 + x + 2^x` that we would still want to collect and combine.